python/grass/jupyter/display.py

```diff
@@ -0,0 +1,58 @@
+# MODULE:    grass.jupyter.display
+#
+# AUTHOR(S): Caitlin Haedrich <caitlin DOT haedrich AT gmail>
+#
+# PURPOSE:   This module contains functions for non-interactive display
+#            in Jupyter Notebooks
+#
+# COPYRIGHT: (C) 2021 Caitlin Haedrich, and by the GRASS Development Team
+#
+#           This program is free software under the GNU General Public
+#           License (>=v2). Read the file COPYING that comes with GRASS
+#           for details.
+
+import os
+from pathlib import Path
+from IPython.display import Image
+import grass.script as gs
+
+
+class GrassRenderer:
+    """The grassRenderer class creates and displays GRASS maps in
+    Jupyter Notebooks."""
+
+    def __init__(
+        self, env=None, width=600, height=400, filename="map.png", text_size=12
+    ):
+        """Initiates an instance of the GrassRenderer class."""
+
+        if env:
+            self._env = env
+        if not env:
+            self._env = os.environ.copy()
+
+        self._env["GRASS_RENDER_WIDTH"] = str(width)
+        self._env["GRASS_RENDER_HEIGHT"] = str(height)
+        self._env["GRASS_TEXT_SIZE"] = str(text_size)
+        self._env["GRASS_RENDER_IMMEDIATE"] = "cairo"
+        self._env["GRASS_RENDER_FILE"] = filename
```

Comment:
Let's get ahead of the game and support `Path` here by converting this to `str` as we do with the numbers. This is how it behaves:

```
> str(Path("/home"))
'/home'
```

Reply:
So, the line would read: `self._env["GRASS_RENDER_FILE"] = str(Path(filename))` ?

Reply:
No, no need to construct Path. The point is to handle the case when user passes filename as an Path object. Try the behavior in an interactive console. The point is to handle these two or three cases:

```python
file = Path("/base/dir") / more_dirs / added_here
renderer1 = GrassRenderer(filename=file)
renderer2 = GrassRenderer(filename="map2.png")
renderer3 = GrassRenderer(filename=os.path.join("base/dir", "map2.png"))
```

Reply:
I couldn't get any of those to produce an error. But, I think I understand: line 38 should read `self._env["GRASS_RENDER_FILE"] = str(filename)`.

Reply:
Right, no error, there is just no point in making the filename a Path object if it is not (well, at least for this line, we need Path later).